src/D2L.CodeStyle.Analyzers/Async/Generator/SyncGenerator.cs

```diff
@@ -167,8 +171,15 @@ CancellationToken cancellationToken
 
 	private static void WriteFiles(
 		SourceProductionContext context,
-		FileGenerationResult result
+		(FileGenerationResult, SyncGeneratorOptions) args 
 	) {
+		var (result, options) = args;
+
+		// TODO: can we read options earlier and not spend time generating?
+		if( options.SuppressSyncGenerator ) {
+			return;
+		}

```

Comment:
I think the easiest spot you could do is `methodsToGenerate.Combine( options ).Select( GenerateSyncMethod )` and update `GenerateSyncMethod` to take `(MethodDeclarationSyntax MethodDeclaration, Compilation Compilation, SyncGeneratorOptions options)`?

Reply:
Ah whoops meant to deal with this before opening the PR.

Ya haha I was writing this in github.dev without IDE support and figured the exact signature for `GenerateSyncMethod` would be nasty so dodged it.

I also wanted to see if I could make it happen earlier, like before `IsInterestingLookingMethod`/`ExtractSyntax` etc. but even with the IDE it wasn't clear, oh well.